de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaTechnologySubGenerator.xtend

```diff
@@ -0,0 +1,181 @@
+            logger.info('''Model written to «targetFolder»''')
+        } else
+            throw new Exception("Technology model generation failed :(")
+        return myTechModel
+    }
+
+    def createProtocols() {
+        //creates "rest" as the default openapi protocol
+        val rest = TECHNOLOGY_FACTORY.createProtocol
+        rest.name = "rest"
+        openApi.paths.forEach[path, pathSchema|searchMediaTimes(pathSchema)]
+        mediaTypes.forEach [ type |
+            val df = TECHNOLOGY_FACTORY.createDataFormat
+            df.formatName = type
+            rest.dataFormats.add(df)
+        ]
+        //Sets "application/json" as default format if the OpenAPI desc entails it at all.
+        // If not, the first encountered media type is set as default.
+        rest.defaultFormat = rest.dataFormats.findFirst[
+            it.formatName.equals("application/json")
+        ] ?: rest.dataFormats.get(0)
+        myTechModel.protocols.add(rest)
+    }
+
+    def searchMediaTimes(PathItem item) {
+        val httpVerbList = <Operation>newArrayList
+        httpVerbList.addAll(item.get, item.delete, item.put,
+            item.post, item.patch, item.head, item.options
+        )
+        httpVerbList.forEach [ verb |
+            if (verb !== null) {
+                // Getting MediaTypes from responses
+                verb.responses?.forEach[
+                    responseCode, responseSchema|addToMediaTypes(responseSchema?.content?.keySet)
+                ]
+                // Getting MediaTypes from requests
+                addToMediaTypes(verb.requestBody?.content?.keySet)
+            }
+
+        ]
+    }
+
+    def addToMediaTypes(Set<String> strings) {
+        if (strings !== null) {
+            strings.forEach[mediaType|mediaTypes.add(mediaType)]
+        }
+    }
+
+    def initilize() {
+        myTechModel.name = OpenApiUtil.removeInvalidCharsFromName("OpenApi")
+        // Adding Primitive Types for OpenApi Technology
+        myTechModel.initPrimitives
+        myTechModel.initAspects
+        // TODO ResponseCodes, Operation Types, Formats
+    }
+
+    /** Creates Default Aspects for HTTP-based communication,
+     * i.e., request methods and response codes.
+     */
+    def void initAspects(Technology technology) {
+        val httpVerbs = <ServiceAspect>newArrayList
+        httpVerbs.add("GetMapping".createServiceAspect)
+        httpVerbs.add("PutMapping".createServiceAspect)
+        httpVerbs.add("PostMapping".createServiceAspect)
+        httpVerbs.add("DeleteMapping".createServiceAspect)
+        httpVerbs.add("OptionsMapping".createServiceAspect)
+        httpVerbs.add("HeadMapping".createServiceAspect)
+        httpVerbs.add("PatchMapping".createServiceAspect)
+        httpVerbs.add("TraceMapping".createServiceAspect)
+        httpVerbs.add("Unspecified".createServiceAspect)
+        myTechModel.serviceAspects.addAll(httpVerbs)
+    }
+
+    def ServiceAspect createServiceAspect(String name) {
```

Comment:
Comment and rename to, e.g., `createMicroserviceOperationAspect` because this is actually the kind of aspect created by the method.